File: src/main/java/me/testaccount666/serversystem/clickablesigns/executables/weather/ActionWeatherSign.java
package me.testaccount666.serversystem.clickablesigns.executables.weather;

import me.testaccount666.serversystem.clickablesigns.AbstractSignClickAction;
import me.testaccount666.serversystem.userdata.User;
import me.testaccount666.serversystem.utils.ChatColor;
import org.bukkit.block.Sign;
import org.bukkit.configuration.file.FileConfiguration;

import static me.testaccount666.serversystem.utils.MessageBuilder.sign;

public class ActionWeatherSign extends AbstractSignClickAction {

    @Override
    public String getBasePermissionNode() {
        return "ClickableSigns.Weather";
    }

    @Override
    protected boolean executeAction(User user, Sign sign, FileConfiguration config) {
        var weatherType = config.getString("WeatherType", sign.getLine(1)).toLowerCase();
        weatherType = ChatColor.stripColor(weatherType);
        if (weatherType.isEmpty()) {
            sign("Weather.NoWeatherSpecified", user).build();
            return false;
        }

        var world = user.getPlayer().getWorld();

        var finalWeatherType = weatherType;
        return switch (weatherType) {
            case "sun", "clear" -> {
                world.setStorm(false);
                world.setThundering(false);
                sign("Weather.WeatherSet", user)
                        .postModifier(message -> message.replace("<WEATHER>", "clear")).build();
                yield true;
            }
            case "rain", "storm" -> {
                world.setStorm(true);
                world.setThundering(false);
                sign("Weather.WeatherSet", user)
                        .postModifier(message -> message.replace("<WEATHER>", "rain")).build();
                yield true;
            }
            case "thunder" -> {
                world.setStorm(true);
                world.setThundering(true);
                sign("Weather.WeatherSet", user)
                        .postModifier(message -> message.replace("<WEATHER>", "thunder")).build();
                yield true;
            }
            default -> {
                sign("Weather.InvalidWeather", user)
                        .postModifier(message -> message.replace("<WEATHER>", finalWeatherType)).build();
                yield false;
            }
        };
    }
}

File: src/main/kotlin/me/testaccount666/serversystem/clickablesigns/executables/weather/ActionWeatherSign.kt
package me.testaccount666.serversystem.clickablesigns.executables.weather

import me.testaccount666.paperktx.colors.ChatColor.Companion.stripColor
import me.testaccount666.serversystem.clickablesigns.AbstractSignClickAction
import me.testaccount666.serversystem.extensions.signMsg
import me.testaccount666.serversystem.userdata.User
import org.bukkit.block.Sign
import org.bukkit.configuration.file.FileConfiguration

class ActionWeatherSign : AbstractSignClickAction() {
    override val basePermissionNode = "ClickableSigns.Weather"

    override fun executeAction(user: User, sign: Sign, config: FileConfiguration, onSuccess: () -> Unit): Boolean {
        var weatherType = config.getString("WeatherType", sign.getLine(1))!!.lowercase()
        weatherType = stripColor(weatherType)
        if (weatherType.isEmpty()) {
            user.signMsg("Weather.NoWeatherSpecified")
            return false
        }

        val world = user.getPlayer()!!.world

        val success = when (weatherType) {
            "sun", "clear" -> {
                world.setStorm(false)
                world.isThundering = false
                user.signMsg("Weather.WeatherSet") {
                    postModifier { it.replace("<WEATHER>", "clear") }
                }
                true
            }

            "rain", "storm" -> {
                world.setStorm(true)
                world.isThundering = false
                user.signMsg("Weather.WeatherSet") {
                    postModifier { it.replace("<WEATHER>", "rain") }
                }
                true
            }

            "thunder" -> {
                world.setStorm(true)
                world.isThundering = true
                user.signMsg("Weather.WeatherSet") {
                    postModifier { it.replace("<WEATHER>", "thunder") }
                }
                true
            }

            else -> {
                user.signMsg("Weather.InvalidWeather") {
                    postModifier { it.replace("<WEATHER>", weatherType) }
                }
                false
            }
        }
        if (success) onSuccess().also { return true }
        return false
    }
}